tcpb/utils.py

```diff
@@ -44,10 +48,52 @@ def atomic_input_to_job_input(atomic_input: AtomicInput) -> pb.JobInput:
         pb.JobInput.ImdOrbitalType,
         atomic_input.keywords.pop("imd_orbital_type", "NO_ORBITAL").upper(),
     )
-
-    # Drop keyword terms already applied to Molecule
-    atomic_input.keywords.pop("charge", None)
-    atomic_input.keywords.pop("spinmult", None)
+    
+    # Second geometry for ci_vec_overlap job
+    # TODO: Not tested yet!
+    try:
+        geom2 = atomic_input.keywords.pop("geom2")
+        if isinstance(geom2, ndarray):
+            geom2 = geom2.flatten()
+        if len(atomic_input.molecule.symbols) != len(geom2) / 3.0:
+            raise ValueError(
+                "Geometry provided to geom2 does not match atom list"
+            )
+        del ji.xyz2[:]
+        ji.xyz2.extend(geom2)
+    except KeyError:
+        pass
+
+    # MM atom geometries
+    try:
+        mm_geom = atomic_input.keywords.pop("mm_geometry")
+    except KeyError:
+        mm_geom = None
+    try:
+        qm_indices = atomic_input.keywords.pop("qm_indices")
+    except KeyError:
+        qm_indices = None
+    try:
+        prmtop = atomic_input.keywords.pop("prmtop")
+    except KeyError:
+        prmtop = None
+    
+    if (mm_geom is None) and (qm_indices is None) and (prmtop is None):
+        # No QMMM
+        pass
+    elif (mm_geom is not None) and (qm_indices is not None) and (prmtop is not None):
+        if isinstance(mm_geom, ndarray):
+            mm_geom = mm_geom.flatten()
+        
+        del ji.mmatom_position[:]
+        del ji.qm_indices[:]
+        ji.mmatom_position.extend(mm_geom)
+        ji.qm_indices.extend(qm_indices)
+        ji.prmtop_path = str(prmtop)
+    else:
+        raise Exception("QM/MM parameters are not set correctly, "
+        "either one of \"mm_geometry\", \"qm_indices\" or \"prmtop\""
+        "was not set!")
```

Comment:
Same here as above comment. Can this be encapsulated in a single function that executes if some `QM/MM` variable is True/False instead of being an implicit statement that executes if certain values are populated on the job input?